lib/mongo/collection/view/readable.rb

```diff
@@ -58,6 +71,7 @@ module Readable
         #
         # @since 2.0.0
         def aggregate(pipeline, options = {})
+          options = @options.slice(*AGGREGATE_OPTIONS).merge(options)
```

Comment:
The AGGREGATE_OPTIONS is an array of strings but the `options` parameter is documented as a Hash with symbol keys. This usage will only work if `options` is actually a BSON::Document.

You can make AGGREGATE_OPTIONS etc. have the values as both strings and symbols or explicitly convert (better in my opinion) or perform way more manipulations here (to avoid forcing `options` into a BSON::Document if it isn't one).